src/StyledOcticon.js

```diff
@@ -1,23 +1,34 @@
-import Octicon from '@primer/octicons-react'
+import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import {COMMON} from './constants'
 import theme from './theme'
 import sx from './sx'
 
-const StyledOcticon = styled(Octicon)`
-  ${COMMON};
-  ${sx};
+function IconWrapper({icon: IconComponent, className, ...rest}) {
+  return (
+    <IconComponent className={className} {...rest}/>
+  )
+}
+
+const StyledOcticon = styled(IconWrapper)`
+  ${COMMON}
+  ${sx}
 `
 
 StyledOcticon.defaultProps = {
-  theme
+  theme,
+  size: 16
 }
 
 StyledOcticon.propTypes = {
   ...COMMON.propTypes,
   ...sx.propTypes,
+  icon: PropTypes.element,
```

Comment:
I think this should be `PropTypes.elementType`, and should be required.

```suggestion
  icon: PropTypes.elementType.isRequired,
```